### PR TITLE
feature: ScatterPlot config

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
First step in #182, "Save Scatter Plot settings to URL (+current open tab)"

*Expected review time: small, 15-20 min*

Creates a new config object for the Scatter Plot tab, containing its current selected feature axes and the range type. This is passed into the `ScatterPlotTab`, and stored as state in the `App`.

The config object will eventually be serialized to the URL for sharing!

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-232/
2. Open the scatter plot tab and click around, changing various settings. Everything should still load in and work as expected.

### CODE REVIEW

File: src/components/Tabs/ScatterPlotTab.tsx [hunk 1]
```diff
@@ -142,26 +139,56 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
...
+    useEffect(() => {
+      if (propValue !== value) {
+        onChange && onChange();
+        startTransition(() => {
+          _setValue(propValue);
+        });
+      }
+    }, [propValue]);
     return [value, setState];
   };
```
Comment (ShrimpCryptid) [inline src/components/Tabs/ScatterPlotTab.tsx:175]: Would like a gut check here on if I made this too complicated, or if I could make the comments clearer.

File: src/components/LoadingSpinner.tsx
Comment (ShrimpCryptid) [inline src/components/LoadingSpinner.tsx:1]: Increased delay before the spinner appears from 0.1 to 0.2 seconds, to reduce flickering

File: src/App.tsx [hunk 2]
```diff
@@ -62,6 +69,10 @@ function App(): ReactElement {
     (current: ViewerConfig, newProperties: Partial<ViewerConfig>) => ({ ...current, ...newProperties }),
     defaultViewerConfig
   );
+  const [scatterPlotConfig, updateScatterPlotConfig] = useReducer(
+    (current: ScatterPlotConfig, newProperties: Partial<ScatterPlotConfig>) => ({ ...current, ...newProperties }),
+    getDefaultScatterPlotConfig()
+  );
```
Comment (ShrimpCryptid) [inline src/App.tsx:75]: The `ViewerConfig` (just above the fold here) uses the same pattern. Using `useReducer` allows `scatterPlotConfig` to be called multiple times in the same state update and reflect all of those changes in the next.

Comment (toloudis) [reply src/App.tsx:75]: do you mean to say "updated multiple times in the same state update"?

Comment (ShrimpCryptid) [reply src/App.tsx:75]: Yes :X

File: src/colorizer/types.ts [hunk 3]
```diff
@@ -167,3 +167,22 @@ export const defaultViewerConfig: ViewerConfig = {
   outOfRangeDrawSettings: { mode: DrawMode.USE_COLOR, color: new Color(OUT_OF_RANGE_COLOR_DEFAULT) },
   outlierDrawSettings: { mode: DrawMode.USE_COLOR, color: new Color(OUTLIER_COLOR_DEFAULT) },
 };
+
+export enum RangeType {
```
Comment (toloudis) [inline src/colorizer/types.ts:171]: Is this going to be another "permanent ids can never be changed" once you serialize this to url?

**Review by toloudis — APPROVED** [review]

File: src/colorizer/types.ts (hunk 3 above)
Comment (ShrimpCryptid) [reply src/colorizer/types.ts:171]: We'll need to always ensure backwards compatibility for these specific values in the URL, but we can always add a layer of indirection here or remap them as needed.

Comment (ShrimpCryptid) [reply src/colorizer/types.ts:171]: See here: https://github.com/allen-cell-animated/nucmorph-colorizer/pull/233/files#diff-49ae85ec948625f41a63acb8324ff556e601766969a1c26ac5c44c28c35ab61aR318-R328

File: src/components/Tabs/ScatterPlotTab.tsx [hunk 4]
```diff
@@ -765,13 +754,15 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
           label={"X"}
           selected={xAxisFeatureName || ""}
           items={menuItems}
-          onChange={setXAxisFeatureName}
+          onChange={(key) => props.updateScatterPlotConfig({ xAxis: key })}
```
Comment (rugeli) [inline src/components/Tabs/ScatterPlotTab.tsx:757]: I like the refactoring of this handler to directly call `props.updateScatterPlotConfig` with the new key!

**Review by rugeli — APPROVED** [review]
Looks good to me! First time reviewing frontend work, it's great to learn how you handled state synchronization and prop updates.

**Pull request merged**